jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/UserConfigurationCompareTest.java

```diff
@@ -0,0 +1,158 @@
+        boolean result = UserConfigurationCompare.configEqual(entity1, entity2);
+        //Then
+        assertTrue(result);
+    }
+
+    @Test
+    public void TestConfigEqualWhenEitherNull() {
+        //Given
+        Object entity1 = null;
+        Object entity2 = "new";
+        //When
+        boolean result = UserConfigurationCompare.configEqual(entity1, entity2);
+        //Then
+        assertFalse(result);
+    }
+
+    @Test
+    public void TestConfigEqualWhenMapAndEqual() {
+        //Given
+        Map<String, String> entity1 = new HashMap();
+        Map<String, String> entity2 = new HashMap();
+        entity1.put("item1", "code");
+        entity2.put("item1", "code");
+        //When
+        boolean result = UserConfigurationCompare.configEqual(entity1, entity2);
+        //Then
+        assertTrue(result);
+    }
+    @Test
+    public void TestConfigEqualWhenMapAndNotEqual() {
+        //Given
+        Map<String, String> entity1 = new HashMap();
+        Map<String, String> entity2 = new HashMap();
+        entity1.put("item", "code");
+        entity2.put("item1", "code");
+        //When
+        boolean result = UserConfigurationCompare.configEqual(entity1, entity2);
+        //Then
+        assertFalse(result);
+    }
+
+
+    @Test
+    public void TestConfigEqualWhenObjectMetaIsTrue() {
+        //Given
+        Object entity1 = new ObjectMetaBuilder().withName("test1").withAnnotations(Collections.singletonMap("foo", "bar")).build();
+        Object entity2 = new ObjectMetaBuilder().withName("test1").withAnnotations(Collections.singletonMap("foo","bar")).build();
+        //When
+        boolean result = UserConfigurationCompare.configEqual(entity1, entity2);
+        //Then
+        assertTrue(result);
+    }
+
+    @Test
+    public void TestConfigEqualWhenCollectionWhenFalse(){
+        //Given
+        ArrayList<String> anandList = new ArrayList<>();
+        Object entity1 = Collections.singletonList(anandList);
+        Object entity2 = Collections.EMPTY_LIST;
+        //When
+        boolean result = UserConfigurationCompare.configEqual(entity1,entity2);
+        //Then
+        assertFalse(result);
+    }
+
+    @Test
+    public void TestConfigEqualWhenCollectionWhenTrue(){
+        //Given
+        Object entity1 = Collections.EMPTY_LIST;
+        Object entity2 = Collections.EMPTY_SET;
+        //When
+        boolean result = UserConfigurationCompare.configEqual(entity1,entity2);
+        //Then
+        assertTrue(result);
+    }
+
+
+    @Test
+    public void TestConfigEqual() {
+        //Given
+        Object entity1 = "asda";
+        Object entity2 = "asdasda";
+        //When
+        boolean result = UserConfigurationCompare.configEqual(entity1, entity2);
+        //Then
+        assertFalse(result);
+    }
+
+    @Test
+    public void TestConfigEqualWhenKDTO() {
+        //Given
+        Object entity1 = new UserConfigurationCompare();
+        Object entity2 = new UserConfigurationCompare();
```

Comment:
Use some Kubernetes object rather than using `UserConfigurationCompare`. Bdw, this class doesn't even have any member field. I'm not sure how your logic would be getting tested.